s2a/src/test/java/io/grpc/s2a/internal/handshaker/S2AProtocolNegotiatorFactoryTest.java

```diff
@@ -158,7 +158,7 @@ public void createProtocolNegotiatorFactory_buildsAnS2AProtocolNegotiatorOnClien
   public void closeProtocolNegotiator_verifyProtocolNegotiatorIsClosedOnClientSide()
       throws Exception {
     InternalProtocolNegotiator.ClientFactory clientFactory =
-        S2AProtocolNegotiatorFactory.createClientFactory(LOCAL_IDENTITY, channelPool);
+        S2AProtocolNegotiatorFactory.createClientFactory(LOCAL_IDENTITY, channelPool, null);
```

Comment:
Shouldn't this, and probably some of the others, pass in the runnable to close the stub and verify that the stub was also closed?

Reply:
These unit tests are not creating the full pipeline, so if I create the stub here and pass it in, we don't hit code where the tls handler gets inserted (thus we don't get to pass the runnable to `InternalProtocolNegotiators.tls`). I added a new Integration test which verifies the stub gets closed. Does that sound ok?

Reply:
That sounds fine.